ddtrace/internal/module.py

```diff
@@ -221,6 +221,21 @@ def __setitem__(self, name, module):
         # type: (str, ModuleType) -> None
         self._modules.__setitem__(name, module)
 
+    @property
+    def _origin_map(self):
+        # type: () -> Dict[str, ModuleType]
+        if self._om is None:
+            try:
+                self._om = {origin(module): module for module in sys.modules.values()}
+            except RuntimeError:
+                # The state of sys.modules might have been mutated by another
+                # thread. We try to build the full mapping at the next occasion.
+                # For now we take the more expensive route of building a list of
+                # the current values, which might be incomplete.
+                return {origin(module): module for module in list(sys.modules.values())}
```

Comment:
we don't set `self._om` here?

```suggestion
                self._om = {origin(module): module for module in list(sys.modules.values())}
```

Reply:
I think the idea is, if `sys.modules` is mutated while trying to iterate over it, then we know it is has changed, so just return a best effort `sys..modules` now, and we'll try again to recompute the next time we access `_origin_map`.

e.g. don't cache the value until we can iterate over `sys.modules` without a mutation error.

Reply:
ah right cause we want a reference to track.

Reply:
Note to self: this previous attempt is in this commit https://github.com/P403n1x87/dd-trace-py/commit/ed5b716526d9a657e474a8db162b257c37e9222e